src/main/java/de/tum/cit/aet/artemis/lecture/web/LectureResource.java

```diff
@@ -201,18 +213,75 @@ public ResponseEntity<Set<Lecture>> getLecturesForCourse(@PathVariable Long cour
      */
     @GetMapping("courses/{courseId}/lectures-with-slides")
     @EnforceAtLeastStudent
-    public ResponseEntity<Set<Lecture>> getLecturesWithSlidesForCourse(@PathVariable Long courseId) {
-        log.debug("REST request to get all Lectures with slides of the units for the course with id : {}", courseId);
+    public ResponseEntity<List<LectureDTO>> getLecturesWithSlidesForCourse(@PathVariable Long courseId) {
+        log.info("Getting all lectures with slides for course {}", courseId);
+        long start = System.currentTimeMillis();
+        var course = courseRepository.findByIdElseThrow(courseId);
+        var user = userRepository.getUserWithGroupsAndAuthorities();
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, user);
```

Comment:
This can be omitted by modifying the annotation to `@EnforceAtLeastStudentInCourse`

Reply:
Good point, in the previous solution, the filtering was different for other users and the user object was still needed below. I'll change it to keep it simple